### PR TITLE
[ASUYTP] Add string JSON field to course settings in cms and payload in courses API

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Added the new field to course settings in cms (Schedule & Details). This is 'extra_json' field.  the field contains JSON and has JSON validation. Available only in cms. 
Also, courses API (api/courses/v1/courses/) has this field in the payload.  API contact: https://docs.google.com/document/d/1UB9SSPdegdxgeRyyylFa8dCPYlQ9U4uDvyN7kEeo59w/edit

**Youtrack:** https://youtrack.raccoongang.com/issue/ASUYT-138

### CODE REVIEW

File: cms/static/js/views/settings/main.js [hunk 1]
```diff
@@ -176,7 +179,8 @@ define(['js/views/validation', 'codemirror', 'underscore', 'jquery', 'jquery.ui'
                    'course_settings_learning_fields': 'course-settings-learning-fields',
                    'add_course_learning_info': 'add-course-learning-info',
                    'add_course_instructor_info': 'add-course-instructor-info',
-                   'course_learning_info': 'course-learning-info'
+                   'course_learning_info': 'course-learning-info',
+                   'extra_json':'extra-json'
```
Comment (idegtiarov) [inline cms/static/js/views/settings/main.js]: please add space after the colon

File: cms/static/js/views/settings/main.js [hunk 2]
```diff
@@ -370,6 +374,41 @@ define(['js/views/validation', 'codemirror', 'underscore', 'jquery', 'jquery.ui'
...
+               codeMirrorsJson: {},
+               codeMirrorizeJson: function(e, forcedTarget) {
+                   var thisTarget, cachethis, field, cmTextArea;
+                   if (forcedTarget) {
+                       thisTarget = forcedTarget;
+                       thisTarget.id = $(thisTarget).attr('id');
+                   } else if (e !== null) {
+                       thisTarget = e.currentTarget;
+                   } else
+        {
```
Comment (idegtiarov) [inline cms/static/js/views/settings/main.js]: Please improve the indentation.

File: openedx/core/djangoapps/models/course_details.py [hunk 3]
```diff
@@ -337,6 +343,17 @@ def recompose_video_tag(video_key):
             )
         return result
 
+    @staticmethod
+    def check_json_string(json_string):
+        """
+        Returns string JSON if json_string is valid json. Return None in another case.
```
Comment (idegtiarov) [inline openedx/core/djangoapps/models/course_details.py]: typo s/Returns/Return/ and s/string JSON/JSON string/ and s/valid json/valid JSON/ and s/in another case/in other cases

File: openedx/core/djangoapps/models/course_details.py [hunk 4]
```diff
@@ -337,6 +343,17 @@ def recompose_video_tag(video_key):
...
+        """
+        try:
+            json.loads(json_string)
+        except ValueError:
+            return None
```
Comment (idegtiarov) [inline openedx/core/djangoapps/models/course_details.py]: None is the default value and could be omitted.

File: cms/static/js/views/settings/main.js (hunk 1 above)
Comment (AndreyLykhoman) [reply cms/static/js/views/settings/main.js]: Done

File: cms/static/js/views/settings/main.js (hunk 2 above)
Comment (AndreyLykhoman) [reply cms/static/js/views/settings/main.js]: Done

File: openedx/core/djangoapps/models/course_details.py (hunk 3 above)
Comment (AndreyLykhoman) [reply openedx/core/djangoapps/models/course_details.py]: Done

File: openedx/core/djangoapps/models/course_details.py (hunk 4 above)
Comment (AndreyLykhoman) [reply openedx/core/djangoapps/models/course_details.py]: Done

**Pull request merged**